pkg/registry/authentication/tokenreview/storage.go

```diff
@@ -67,20 +71,31 @@ func (r *REST) Create(ctx context.Context, obj runtime.Object, createValidation
 	fakeReq := &http.Request{Header: http.Header{}}
 	fakeReq.Header.Add("Authorization", "Bearer "+tokenReview.Spec.Token)
 
+	auds := tokenReview.Spec.Audiences
+	if len(tokenReview.Spec.Audiences) == 0 {
+		auds = r.apiAudiences
+	}
+
+	fakeReq = fakeReq.WithContext(genericapirequest.WithAudiences(fakeReq.Context(), auds))
+
 	resp, ok, err := r.tokenAuthenticator.AuthenticateRequest(fakeReq)
 	tokenReview.Status.Authenticated = ok
 	if err != nil {
 		tokenReview.Status.Error = err.Error()
 	}
```

Comment:
should we be requiring the response to be non-nil, and for resp.Status.Audience to be one of the audiences in tokenReview.Spec.Audiences here? or are we counting on the authenticator impl to do that for us?